modal/image.py

```diff
@@ -553,6 +624,23 @@ def build_dockerfile(version: ImageBuilderVersion) -> DockerfileSpec:
             context_mount=mount,
         )
 
+    def add_local_python_packages(self, *packages: Union[str, Path], copy: bool = False) -> "_Image":
+        """Attaches local Python packages to the container running the image
+
+        Packages are added to the /root directory which is on the PYTHONPATH of any
+        executed Modal functions.
+
+        By default (copy=False) the packages are layered on top of the image when
+        the container starts up and not built as an image layer.
```

Comment:
Rewrote:

Adds Python Package Files to Containers

Adds all files from the specified Python packages to containers running the image.

Packages are added to the `/root` directory, which is on the `PYTHONPATH` of any executed Modal functions.

By default (`copy=False`), the files are added to containers on startup and are not built into the actual image, which speeds up deployment.

Set `copy=True` to copy the files into an image layer at build time instead. This can slow down iteration since it requires a rebuild of the image and any subsequent build steps whenever the included files change, but it is required if you want to run additional build steps after this one.

**Note:** This excludes all dot-prefixed subdirectories or files and all `.pyc`/`__pycache__` files. To add full directories with finer control, use `.add_local_dir()` instead.